heron/scheduler-core/src/java/com/twitter/heron/scheduler/RuntimeManagerRunner.java

```diff
@@ -159,23 +207,25 @@ protected boolean cleanState(
 
     Boolean result;
 
-    // It is possible that  TMasterLocation, PhysicalPlan and SchedulerLocation are not set
-    // Just log but don't consider them failure
+    // It is possible that TMasterLocation, PackingPlan, PhysicalPlan and SchedulerLocation are not
+    // set. Just log but don't consider it a failure and don't return false
     result = statemgr.deleteTMasterLocation(topologyName);
     if (result == null || !result) {
-      // We would not return false since it is possible that TMaster didn't write physical plan
       LOG.warning("Failed to clear TMaster location. Check whether TMaster set it correctly.");
     }
 
+    result = statemgr.deletePackingPlan(topologyName);
+    if (result == null || !result) {
+      LOG.warning("Failed to clear packing plan. Check whether TMaster set it correctly.");
```

Comment:
Fixed. Let's go with this change and cancel #1348 to keep the relevant updates together.